presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCacheManager.java

```diff
@@ -81,11 +84,16 @@
 
     private final ExecutorService cacheFlushExecutor;
     private final ExecutorService cacheRemovalExecutor;
+    private final ScheduledExecutorService cacheSizeCalculateExecutor;
 
     // a mapping from remote file `F` to a range map `M`; the corresponding local cache file for each range in `M` represents the cached chunk of `F`
     private final Map<Path, CacheRange> persistedRanges = new ConcurrentHashMap<>();
     // a local cache only to control the lifecycle of persisted
-    private final Cache<Path, Boolean> cache;
+    // Path and its corresponding cacheScope identifier
+    private final Cache<Path, Long> cache;
+    // CacheScopeHandle identifier to its cached files mapping
+    private final Map<Long, Set<Path>> cacheScopeFiles = new ConcurrentHashMap<>();
```

Comment:
You're right, here, the cache capacity & quota calculation would be an estimation, like I said it would an experimental feature, it would be served more as an evaluation and provide basic protection from bad table data abusing the system